specs/APIReview_TextureStream.md

```diff
@@ -0,0 +1,592 @@
+  /// for creating the D3D Device.
+  HRESULT CreateTextureStream(
+      [in] LPCWSTR streamId,
+      [in] IUnknown* d3dDevice,
+      [out, retval ] ICoreWebView2StagingTextureStream** value);
+  /// Get the graphics adapter LUID of the renderer. The host should use this
+  /// LUID adapter when creating D3D device to use with CreateTextureStream().
+  [propget] HRESULT RenderAdapterLUID([out, retval] LUID* luid);
+  /// Listens for change of graphics adapter LUID of the browser.
+  /// The host can get the updated LUID by RenderAdapterLUID. It is expected
+  /// that the host updates texture's d3d Device with UpdateD3DDevice,
+  /// removes existing buffers and creates new buffer.
+  HRESULT add_RenderAdapterLUIDChanged(
+      [in] ICoreWebView2StagingRenderAdapterLUIDChangedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove listener for start stream request.
+  HRESULT remove_RenderAdapterLUIDChanged(
+      [in] EventRegistrationToken token);
+}
+/// This is the interface that handles texture streaming.
+[uuid(afca8431-633f-4528-abfe-7fc3bedd8962), object, pointer_default(unique)]
+interface ICoreWebView2StagingTextureStream : IUnknown {
+  /// Get the stream ID of the object that is used when calling CreateTextureStream.
+  /// The caller must free the returned string with CoTaskMemFree. See
+  /// [API Conventions](/microsoft-edge/webview2/concepts/win32-api-conventions#strings).
+  // MSOWNERS: TBD (wv2core@microsoft.com)
+  [propget] HRESULT StreamId([out, retval] LPWSTR* id);
+  /// Adds an allowed url origin for the given stream id. The stream requests
+  /// could be made from any frame, including iframes, but these origins
+  /// must be registered first in order for the request to succeed.
+  /// The added filter will be persistent until
+  /// ICoreWebView2StagingTextureStream is destroyed or
+  /// RemoveAllowedOrigin is called.
+  /// The renderer does not support wildcard so it will compare
+  /// literal string input to the requesting frame origin. So, the input string
+  /// should have a scheme like https://.
+  /// For example, https://www.valid-host.com, http://www.valid-host.com are
+  /// valid origins but  www.valid-host.com, or *.valid-host.com. are not
+  /// valid origins.
+  /// getTextureStream() will fail unless the requesting frame's origin URL is
+  /// added to the request filter.
+  HRESULT AddAllowedOrigin([in] LPCWSTR origin);
+  /// Remove added origin, which was added by AddAllowedOrigin.
+  HRESULT RemoveAllowedOrigin([in] LPCWSTR origin);
+  /// Listens for stream requests from the Javascript's getTextureStream call
+  /// for this stream's id. It is called for the first request only, and will
+  /// not be called with subsequent requests of same stream id from any pages
+  /// in the middle of request handling or after it returns success.
+  /// The request is regarded as success only when the host provides the stream,

```

Comment:
If we look at in terms of state, yes it has only Start and Stop state.  Javascript side, it will follow the existing MediaStream lifetime, which is from w3c standard. https://www.w3.org/TR/mediacapture-streams/ AS the return object is MediaStream on JS side.

  /// The Texture stream becomes 'Started' state once it starts sending a texture
  /// and becomes 'Stopped' state on the moment when it receives StopRequested.

Reply:
"What happens when native calls Present but JS never called Start? " 
It is never started. (Present call (within 10s) after StartRequested is valid).

"Is the texture stream 'started' and if JS tries to call 'Start' it will just immediately start receiving what the texture stream has already been sending"
No with above reason.